waterdata/tests/test_views.py

```diff
@@ -46,25 +44,83 @@ def setUp(self):
+                               '-77.12763889\tS\tNAD83\t 37.20\t .1\tNAVD88\t02070008\tuv\t00010\t\t69932\t'
+                               '7.1 ft from riverbed (top)\twat\t\t1645597\t0\t2007-10-01\t2018-01-10\t3754\nUSGS\t'
+                               '01630500\tPOTOMAC RIVER NEAR WASH, DC LITTLE FALLS PUMP STA\tST\t38.94977778\t'
+                               '-77.12763889\tS\tNAD83\t 37.20\t .1\tNAVD88\t02070008\tuv\t00010\t\t69942\t'
+                               'From multiparameter sonde\twat\t\t1645597\t0\t2013-11-23\t2018-01-10\t1509\nUSGS\t'
+                               '01630500\tPOTOMAC RIVER NEAR WASH, DC LITTLE FALLS PUMP STA\tST\t38.94977778\t'
+                               '-77.12763889\tS\tNAD83\t 37.20\t .1\tNAVD88\t02070008\tuv\t00060\t\t69928\t\twat\t\t'
+                               '1645423\t0\t1972-06-09\t2018-01-10\t16651\nUSGS\t01630500\t'
+                               'POTOMAC RIVER NEAR WASH, DC LITTLE FALLS PUMP STA\tST\t38.94977778\t'
+                               '-77.12763889\tS\tNAD83\t 37.20\t .1\tNAVD88\t02070008\tuv\t00065\t\t69929\t\twat\t\t'
+                               '17164583\t0\t2007-10-01\t2018-01-10\t3754\nUSGS\t01630500\t'
+                               'POTOMAC RIVER NEAR WASH, DC LITTLE FALLS PUMP STA\tST\t38.94977778\t-77.12763889\tS\t'
+                               'NAD83\t 37.20\t .1\tNAVD88\t02070008\tuv\t00095\t\t69933\t7.1 ft from riverbed (top)\t'
+                               'wat\t\t1646694\t0\t2007-10-01\t2018-01-10\t3754\nUSGS\t01630500\t'
+                               'POTOMAC RIVER NEAR WASH, DC LITTLE FALLS PUMP STA\tST\t38.94977778\t-77.12763889\tS\t'
+                               'NAD83\t 37.20\t .1\tNAVD88\t02070008\tuv\t00095\t\t69943\tFrom multiparameter sonde\t'
+                               'wat\t\t1646694\t0\t2013-11-23\t2018-01-10\t1509'
+                               )
+        self.parameter_lines = self.test_param_rdb.split('\n')
 
-    @requests_mock.mock()
+    @mock.patch('waterdata.views.execute_get_request')
     def test_everything_okay(self, r_mock):
-        r_mock.get(self.test_url, status_code=200, text=self.test_rdb_text)
-        response = self.app_client.get('/monitoring-location/{}'.format(self.test_site_number))
+        m_resp = mock.Mock()
+        m_resp.status_code = 200
+        m_resp.text = self.test_rdb_text
+        m_resp.iter_lines.return_value = iter(self.test_rdb_lines)
+        m_resp_param = mock.Mock()
+        m_resp_param.status_code = 200
+        m_resp_param.iter_lines.return_value = iter(self.parameter_lines)
+
+        return_values = [m_resp, m_resp_param]
+        r_mock.side_effect = return_values
+
+        response = self.app_client.get('/monitoring-location/{}?agency_cd=USGS'.format(self.test_site_number))
         self.assertEqual(response.status_code, 200)
         self.assertIn('Some Random Site', response.data.decode('utf-8'))
+        self.assertIn('@context', response.data.decode('utf-8'))
```

Comment:
This is just something I've been think about as far as testing views. What do you think of mocking the call to render_template rather than check the contents of the template? It would remove the dependency on the template. Just looking for your opinion.

Reply:
From a unit test perspective, I think it makes sense because then we're just testing the view code and making sure that `render_template` is being called with the parameters we expect.

On the other-hand, there it's nice to also be able to test the logic in the jinja templates. Perhaps what that really get's at is that we should try to unit test Jinja templates as discussed here.

So, I'd lean slightly towards mocking `render_template` and writing up distinct tests for anything we might want to test in a template.